apiserver/common/networkingcommon/networkconfigapi.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// The networkconfigapi package implements the network config parts
+// common to machiner and provisioner interface
+
+package networkingcommon
+
+import (
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/errors"
+	"github.com/juju/juju/apiserver/common"
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/state"
+	"github.com/juju/juju/state/stateenvirons"
+)
+
+type NetworkConfigAPI struct {
+	st           *state.State
+	getCanModify common.GetAuthFunc
+	getAuthFunc  common.GetAuthFunc
+}
+
+func NewNetworkConfigAPI(st *state.State, getCanModify common.GetAuthFunc) *NetworkConfigAPI {
+	return &NetworkConfigAPI{
+		st:           st,
+		getCanModify: getCanModify,
+	}
+}
+
+func (api *NetworkConfigAPI) getMachine(tag names.Tag) (*state.Machine, error) {
+	entity, err := api.st.FindEntity(tag)
```

Comment:
You're taking a 'names.Tag' and assuming that FindEntity will return a state.Machine type. However, Tag can be a Unit or lots of other things.
If you were taking a names.MachineTag I'd be a bit more ok with a bare ".(*state.Machine)" type assertion, but generally I'd be much happier with doing a 
machine, ok := entity.(*state.Machine)